compiler/qsc_frontend/src/resolve.rs

```diff
@@ -463,20 +501,37 @@ impl Resolver {
         }
     }
 
-    fn bind_open(&mut self, name: &ast::Ident, alias: &Option<Box<ast::Ident>>) {
-        let alias = alias.as_ref().map_or("".into(), |a| Rc::clone(&a.name));
-        if self.globals.namespaces.contains(&name.name) {
+    fn bind_open(&mut self, name: &Idents, alias: &Option<Box<Ident>>) {
+        let Some(id) = self.globals.find_namespace(name.str_iter()) else {
+            self.errors.push(Error::NotFound(
+                name.iter()
+                    .map(|x| x.name.to_string())
+                    .collect::<Vec<_>>()
+                    .join("."),
+                name.span(),
+            ));
```

Comment:
Use the `name.name()` implementation instead of manually joining?

```suggestion
            self.errors
                .push(Error::NotFound(name.name().to_string(), name.span()));
```